File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.js
/**
 * @param {string} s
 * @param {number} k
 * @return {number}
 */
var characterReplacement = function(s, k) {
  const count = new Array(26).fill(0);
  let [l, r, maxCount] = [0, 0, 0];
  
  while (r < s.length) {
    maxCount = Math.max(maxCount, ++count[s.charCodeAt(r) - 'A'.charCodeAt()]);
    while (r - l + 1 - maxCount > k) 
      count[s.charCodeAt(l++) - 'A'.charCodeAt()]--;
    
    r++;
  }
  
  return r - l;
};

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        count = Counter()
        j = max_cnt = 0
        for i, ch in enumerate(s):
            count[ch] += 1
            max_cnt = max(max_cnt, count[ch])
            while i - j + 1 - max_cnt > k:
                count[s[j]] -= 1
                j += 1
        return len(s) - j